src/components/themed/PromoBanner.js

```diff
@@ -0,0 +1,49 @@
+// @flow
+
+import * as React from 'react'
+
+import { hideMessageTweak } from '../../actions/AccountReferralActions.js'
+import { linkReferralWithCurrencies } from '../../actions/WalletListActions.js'
+import { useMemo } from '../../types/reactHooks.js'
+import { useDispatch, useSelector } from '../../types/reactRedux.js'
+import { bestOfMessages } from '../../util/ReferralHelpers.js'
+import { PromoCard } from './PromoCard.js'
+
+export const PromoBanner = () => {
+  const dispatch = useDispatch()
+
+  const accountMessages = useSelector(state => state.account.referralCache.accountMessages)
+  const accountReferral = useSelector(state => state.account.accountReferral)
+
+  // Look for active referral promotion
+  const referralPromoCard = useMemo(() => {
```

Comment:
Don't use 'useMemo' to bundle multiple things into an object.

You can call this outside of the memo: 
`const messageSummary = bestOfMessages(accountMessages, accountReferral)`

And you also don't need to memoize the 'message' and 'iconUri' since they are strings.


Which leaves you with just two 'useCallbacks' for `onPress` and `onClose`.